pkg/gardenlet/controller/networkpolicy/reconciler.go

```diff
@@ -348,14 +348,16 @@ func (r *Reconciler) reconcileNetworkPolicyAllowToPrivateNetworks(ctx context.Co
 			return err
 		}
 
-		if v := shoot.Spec.Networking.Nodes; v != nil {
-			blockedNetworkPeers = append(blockedNetworkPeers, *v)
-		}
-		if v := shoot.Spec.Networking.Pods; v != nil {
-			blockedNetworkPeers = append(blockedNetworkPeers, *v)
-		}
-		if v := shoot.Spec.Networking.Services; v != nil {
-			blockedNetworkPeers = append(blockedNetworkPeers, *v)
+		if !shoot.IsWorkerless() {
+			if v := shoot.Spec.Networking.Nodes; v != nil {
+				blockedNetworkPeers = append(blockedNetworkPeers, *v)
+			}
+			if v := shoot.Spec.Networking.Pods; v != nil {
+				blockedNetworkPeers = append(blockedNetworkPeers, *v)
+			}
+			if v := shoot.Spec.Networking.Services; v != nil {
```

Comment:
Services can be non nil for workerless shoots. Should this be inside the condition for shoots with workers only?